### PR TITLE
feat: introduces experimental `useCurrentSession` hook

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { ProfileId } from '../../entities';
+import type { LogoutReason } from '../wallets';
+
+/**
+ * The identifier of a Profile
+ */
+export type ProfileIdentifier = {
```
Comment (desfero) [inline packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts:7]: Any specific reason to not use `Profile` entity?

File: packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  id: ProfileId;
+  handle: string;
+};
+
+/**
+ * Framework-agnostic Wallet representation
+ */
+export type WalletData = {
+  address: string;
```
Comment (desfero) [inline packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts:16]: nit: could use `EthereumAddress` from `shared-kernel` which we could brand at some point in time

File: packages/api-bindings/src/apollo/cache/createProfileTypePolicy.ts [hunk 3]
```diff
@@ -73,9 +71,9 @@ export function createProfileTypePolicy(
 
       followStatus: {
         read(_, { readField }) {
-          const activeWallet = activeWalletVar();
+          const session = getSession();
 
-          if (!activeWallet) {
+          if (!session || session.isNotAuthenticated()) {
```
Comment (desfero) [inline packages/api-bindings/src/apollo/cache/createProfileTypePolicy.ts:76]: what could be the use case where `getSession` return a null value? 
Should we make it an `invariant` as it should never happen if the app is properly wired together?

**Review by desfero — APPROVED** [review]

File: packages/api-bindings/src/apollo/cache/createProfileTypePolicy.ts (hunk 3 above)
Comment (cesarenaldi) [reply packages/api-bindings/src/apollo/cache/createProfileTypePolicy.ts:76]: If everything is wired correctly requests requiring the session should never fire. Maybe it's a big assumption that all hooks (present and future) will be bug free from day-one.

These invariant are difficult to catch unless we increase our test coverage (not all read hooks do have an integration unit test). Also these field policies are triggered by profile/publication requests done in presenters. I would need to think about it if it's possible for those to fire before we know current session state.

File: packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts (hunk 2 above)
Comment (cesarenaldi) [reply packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts:16]: good call

File: packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts (hunk 1 above)
Comment (cesarenaldi) [reply packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts:7]: Passing a DTO to the presenter, not the entity directly. Leveraging TS structural typing to not create a new object until we need to.

**Pull request merged**